File: Data/DTOs/ProjectDTO.cs
namespace Data.DTOs;

public class ProjectDTO
{
    public int ProjectId { get; set; }

    public string ProjectNumber { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public DateTime StartDate { get; set; }
    public DateTime? EndDate { get; set; }
    public int StatusId { get; set; }
    public string? StatusName { get; set; }
    public int CustomerId { get; set; }
    public string? CustomerName { get; set; }
    public int ServiceId { get; set; }
    public string? ServiceName { get; set; }
    public int ProjectManagerId { get; set; }
    public string? ProjectManagerName { get; set; }
    public decimal TotalPrice { get; set; }
}

File: frontend/src/types/ProjectDTO.ts
export interface Status {
    statusId: number;
    statusName: string;
}

export interface Customer {
    customerId: number;
    customerName: string;
    contactPerson: string;
}

export interface Service {
    serviceId: number;
    serviceName: string;
    pricePerUnit: number;
    serviceTypeId: number;
}

export interface ProjectManager {
    projectManagerId: number;
    firstName: string;
    lastName: string;
    email: string;
}

export interface ProjectDTO {
    projectNumber: string;
    projectId: number;
    name: string;
    description: string;
    startDate: string;
    endDate?: string;
    status?: Status;
    customer: Customer;
    service: Service;
    projectManager: ProjectManager;
    totalPrice: number;
}